app/lib/features/deep_linking/parse_acter_uri.dart

```diff
@@ -32,14 +40,41 @@ UriParseResult _parseActerUri(Uri uri) {
   };
 }
 
+UriParseResult _parseHttpsUri(Uri uri) {
+  if (uri.host == 'matrix.to') {
+    return _parseMatrixHttpsUri(uri);
+  }
+  final hash = uri.pathSegments.lastOrNull;
+  if (hash == null || hash.isEmpty) {
+    throw IncorrectHashError();
+  }
+
+  final pathWithoutHash = uri.path.substring(0, uri.path.length - hash.length -1);
+  final strippedUri = uri.replace(path: pathWithoutHash);
+  final hashableUri = strippedUri.toString();
+  final calculatedHash = sha1.convert(utf8.encode(hashableUri)).toString();
+  if (calculatedHash != hash) {
+    throw IncorrectHashError();
+  }
```

Comment:
1. look for the hash
2. remove the hash
3. compare the hash

Reply:
- - -
Example: `acter:roomid/room:acter.global/e/someEvent?via=acter.global&via=example.org` becomes `https://app.acter.global/p/$HASH?via=acter.global&via=example.org#roomid/room:acter.global/e/someEvent` (where `$HASH = SHA1.of(https://app.acter.global/p/?via=acter.global&via=example.org#roomid/room:acter.global/e/someEvent`)
- - -
Not quite understand link structure and logic. Sorry but looks complex to me to understand.

Reply:
Well, not sure what to say. it is _complex_ by nature, as we try to have a link contain all the information of itself in a hash. It is just a question of how to do the pattern. Not sure what else to say here... other than:

```dart
final url = 'acter:roomid/room:acter.global/e/someEvent?via=acter.global&via=example.org'
final scheme = url.scheme; // = 'acter'
final path = url.path; // = 'roomid/room:acter.global/e/someEvent'
final query = url.query; // = 'via=acter.global&via=example.org'

final targetUriPrefix = 'https://app.acter.global/p/';

// which is then constructed a new as
final hash = SHA1.of('$targetUriPrefix?$query#$path'); // = '0c51512e60831746f59b479bc7371693b15f0ad0'

and then becomes:
final finalUri = `$targetUriPrefix$hash?$query#$path` // = 'https://app.acter.global/p/0c51512e60831746f59b479bc7371693b15f0ad0?via=acter.global&via=example.org#roomid/room:acter.global/e/someEvent'

```